UI  startup

### Summary

Not an expert in this field but we are getting a lot of warnings when the UI is starting:

```
./node_modules/direktiv-react-hooks/dist/index.js
Module not found: Can't resolve 'encoding' in '/home/direktiv/ui-december-2021/node_modules/direktiv-react-hooks/dist'

src/App.js
  Line 19:12:  'err' is assigned a value but never used  no-unused-vars

src/components/diagram/index.js
  Line 16:17:  Unexpected mix of '&&' and '||'                                                                             no-mixed-operators
  Line 16:38:  Unexpected mix of '&&' and '||'                                                                             no-mixed-operators
  Line 70:7:   React Hook useEffect has a missing dependency: 'ostatus'. Either include it or remove the dependency array  react-hooks/exhaustive-deps

src/components/editor/index.js
  Line 2:10:   'linkHorizontal' is defined but never used                                                                                 no-unused-vars
  Line 290:7:  React Hook useEffect has missing dependencies: 'dlang' and 'validate'. Either include them or remove the dependency array  react-hooks/exhaustive-deps

src/components/loader/index.js
  Line 27:7:  React Hook useEffect has missing dependencies: 'display' and 'timer'. Either include them or remove the dependency array  react-hooks/exhaustive-deps

src/components/modal/index.js
  Line 117:35:  Expected '!==' and instead saw '!='  eqeqeq

src/components/navbar/index.js
  Line 9:56:   'VscLock' is defined but never used           no-unused-vars
  Line 9:76:   'VscSettings' is defined but never used       no-unused-vars
  Line 9:106:  'VscSignOut' is defined but never used        no-unused-vars
  Line 16:34:  'akeyReq' is assigned a value but never used  no-unused-vars

src/components/sankey/index.js
  Line 8:8:  'objectHash' is defined but never used  no-unused-vars

src/layouts/events/index.js
  Line 5:68:   'ContentPanelHeaderButtonIcon' is defined but never used  no-unused-vars
  Line 42:10:  'errHistory' is assigned a value but never used           no-unused-vars
  Line 42:22:  'errListeners' is assigned a value but never used         no-unused-vars

src/layouts/explorer/index.js
  Line 6:10:  'VscTriangleDown' is defined but never used  no-unused-vars
  Line 8:10:  'FiEdit' is defined but never used           no-unused-vars

src/layouts/explorer/workflow/index.js
  Line 70:18:    'err' is assigned a value but never used      no-unused-vars
  Line 75:12:    'revsErr' is assigned a value but never used  no-unused-vars
  Line 115:542:  No duplicate props allowed                    react/jsx-no-duplicate-props
  Line 115:632:  No duplicate props allowed                    react/jsx-no-duplicate-props

src/layouts/explorer/workflow/revisionTab.js
  Line 497:77:  'isTag' is assigned a value but never used  no-unused-vars

src/layouts/instance/index.js
  Line 6:42:     'ContentPanelFooter' is defined but never used       no-unused-vars
  Line 13:60:    'WindowScroller' is defined but never used           no-unused-vars
  Line 43:19:    'setWidth' is assigned a value but never used        no-unused-vars
  Line 49:21:    'cancelInstance' is assigned a value but never used  no-unused-vars
  Line 137:205:  No duplicate props allowed                           react/jsx-no-duplicate-props
  Line 142:183:  No duplicate props allowed                           react/jsx-no-duplicate-props
  Line 420:10:   'InstanceTuple' is defined but never used            no-unused-vars

src/layouts/instances/index.js
  Line 16:10:   'path' is defined but never used                      no-unused-vars
  Line 38:26:   'setIQueryParams' is assigned a value but never used  no-unused-vars
  Line 39:23:   'pageInfo' is assigned a value but never used         no-unused-vars
  Line 124:36:  'startedFrom' is assigned a value but never used      no-unused-vars
  Line 124:59:  'finishedFrom' is assigned a value but never used     no-unused-vars
  Line 158:28:  Expected '!==' and instead saw '!='                   eqeqeq

src/layouts/main/index.js
  Line 24:8:   'Login' is defined but never used          no-unused-vars
  Line 32:90:  'akey' is assigned a value but never used  no-unused-vars

src/layouts/monitoring/index.js
  Line 3:10:  'VscGraph' is defined but never used  no-unused-vars

src/layouts/namespace-services/pod.js
  Line 151:11:  'params' is assigned a value but never used  no-unused-vars

src/layouts/settings/broadcast-panel/index.js
  Line 4:10:  'BsSliders' is defined but never used  no-unused-vars

src/layouts/settings/variables-panel/index.js
  Line 12:9:   'BsUpload' is defined but never used          no-unused-vars
  Line 14:10:  'RiDeleteBin2Line' is defined but never used  no-unused-vars
```

Is that normal or can/should that be fixed?

--- a/package.json
+++ b/package.json
@@ -12,7 +12,7 @@
     "d3-sankey-circular": "^0.34.0",
     "dagre": "^0.8.5",
     "dayjs": "^1.10.7",
-    "direktiv-react-hooks": "^0.0.137",
+    "direktiv-react-hooks": "^0.0.143",
     "js-yaml": "^4.1.0",
     "object-hash": "^2.2.0",
     "rc-slider": "^9.7.5",
@@ -55,5 +55,8 @@
       "last 1 firefox version",
       "last 1 safari version"
     ]
+  },
+  "devDependencies": {
+    "encoding": "^0.1.13"
   }
 }

--- a/src/App.js
+++ b/src/App.js
@@ -16,7 +16,8 @@ function App() {
     const [akey, setAKey] = useState(localStorage.getItem('apikey'))
     const [login, setLogin] = useState(false)
     const [akeyReq, setAKeyReq] = useState(false)
-    const [err ,setErr] = useState("")
+    // Todo find nice way to handle error
+    const [,setErr] = useState("")
 
     useEffect(()=>{
         async function fetchVersion() {

--- a/src/components/diagram/index.js
+++ b/src/components/diagram/index.js
@@ -13,7 +13,7 @@ export default function WorkflowDiagram(props) {
     const [ostatus, setOStatus] = useState(instanceStatus)
 
     useEffect(()=>{
-        if(load && workflow !== null || instanceStatus !== ostatus) {
+        if(load && (workflow !== null || instanceStatus !== ostatus)) {
             // initialize the dagre graph
             const dagreGraph = new dagre.graphlib.Graph()
             dagreGraph.setDefaultEdgeLabel(() => ({}))
@@ -67,7 +67,7 @@ export default function WorkflowDiagram(props) {
             setLoad(false)
         }
         
-    },[load, workflow,flow, instanceStatus])
+    },[load, workflow,flow, instanceStatus, ostatus])
 
     if(load) {
         return <></>

--- a/src/components/editor/index.js
+++ b/src/components/editor/index.js
@@ -1,5 +1,4 @@
 import Editor, {useMonaco} from "@monaco-editor/react";
-import { linkHorizontal } from "d3-shape";
 import { useEffect } from "react";
 import './style.css'
 // import * as cobalt from './cobalt.json'
@@ -287,7 +286,7 @@ export default function DirektivEditor(props) {
             // })
         }
         // monaco.editor.layout()
-    },[monaco])
+    },[monaco, dlang, validate])
 
     function handleEditorChange(value, event) {
         setDValue(value)

--- a/src/components/modal/index.js
+++ b/src/components/modal/index.js
@@ -114,7 +114,7 @@ function ModalOverlay(props) {
                 const action = keyDownActions[i];
 
                 let fn = async (e) => {
-                    if (action.id != undefined && action.id !== e.target.id){
+                    if (action.id !== undefined && action.id !== e.target.id){
                         return
                     }
 

--- a/src/components/navbar/index.js
+++ b/src/components/navbar/index.js
@@ -6,14 +6,14 @@ import NamespaceSelector from '../namespace-selector';
 
 import Modal, { KeyDownDefinition } from '../modal';
 import { ButtonDefinition } from '../modal';
-import {VscAdd,  VscFolderOpened, VscGraph, VscLayers, VscLock, VscServer, VscSettings, VscSettingsGear, VscSignOut, VscSymbolEvent, VscVmRunning} from 'react-icons/vsc';
+import {VscAdd,  VscFolderOpened, VscGraph, VscLayers, VscServer,  VscSettingsGear,  VscSymbolEvent, VscVmRunning} from 'react-icons/vsc';
 
 import {IoExtensionPuzzleOutline} from 'react-icons/io5';
 import { Link, matchPath, useLocation, useNavigate } from 'react-router-dom';
 
 function NavBar(props) {
 
-    let {onClick, style, footer, akeyReq, className, createNamespace, namespace, namespaces, createErr, toggleResponsive, setToggleResponsive, extraNavigation} = props;
+    let {onClick, style, footer,  className, createNamespace, namespace, namespaces, createErr, toggleResponsive, setToggleResponsive, extraNavigation} = props;
     if (!className) {
         className = ""
     }

--- a/src/components/sankey/index.js
+++ b/src/components/sankey/index.js
@@ -5,7 +5,6 @@ import * as d3 from 'd3'
 import { sankeyCircular, sankeyJustify } from 'd3-sankey-circular'
 import {GenerateRandomKey} from '../../util';
 
-import objectHash from 'object-hash';
 
 export default function Sankey(props) {
     const {getWorkflowSankeyMetrics, revision} = props

--- a/src/layouts/events/index.js
+++ b/src/layouts/events/index.js
@@ -2,7 +2,7 @@ import React, { useState } from 'react';
 import { Config } from '../../util';
 import {  VscCloud, VscRedo, VscSymbolEvent } from 'react-icons/vsc';
 import Button from '../../components/button';
-import ContentPanel, { ContentPanelBody, ContentPanelHeaderButton, ContentPanelHeaderButtonIcon, ContentPanelTitle, ContentPanelTitleIcon } from '../../components/content-panel';
+import ContentPanel, { ContentPanelBody, ContentPanelHeaderButton, ContentPanelTitle, ContentPanelTitleIcon } from '../../components/content-panel';
 import FlexBox from '../../components/flexbox';
 import {useEvents} from 'direktiv-react-hooks'
 import Modal, { ButtonDefinition } from '../../components/modal';
@@ -39,7 +39,8 @@ function EventsPage(props) {
 
     let {namespace} = props;
 
-    let {errHistory, errListeners, eventHistory, eventListeners, sendEvent, replayEvent} = useEvents(Config.url, true, namespace, localStorage.getItem("apikey"))
+    // errHistory and errListeners TODO show error if one
+    let {eventHistory, eventListeners, sendEvent, replayEvent} = useEvents(Config.url, true, namespace, localStorage.getItem("apikey"))
 
     return(
         <>

--- a/src/layouts/explorer/index.js
+++ b/src/layouts/explorer/index.js
@@ -3,9 +3,9 @@ import './style.css';
 
 import ContentPanel, { ContentPanelBody, ContentPanelHeaderButton, ContentPanelHeaderButtonIcon, ContentPanelTitle, ContentPanelTitleIcon } from '../../components/content-panel';
 import FlexBox from '../../components/flexbox';
-import { VscTriangleDown, VscAdd, VscClose,  VscSearch, VscEdit, VscTrash, VscFolderOpened } from 'react-icons/vsc';
+import { VscAdd, VscClose,  VscSearch, VscEdit, VscTrash, VscFolderOpened } from 'react-icons/vsc';
 import { Config, GenerateRandomKey } from '../../util';
-import { FiEdit, FiFolder } from 'react-icons/fi';
+import { FiFolder } from 'react-icons/fi';
 import { FcWorkflow } from 'react-icons/fc';
 import { HiOutlineTrash } from 'react-icons/hi';
 import { useNodes } from 'direktiv-react-hooks';

--- a/src/layouts/explorer/workflow/index.js
+++ b/src/layouts/explorer/workflow/index.js
@@ -73,12 +73,14 @@ function InitialWorkflowHook(props){
 
     const [activeTab, setActiveTab] = useState(searchParams.get("tab") !== null ? parseInt(searchParams.get('tab')): 0)
 
-    const {data, err, getSuccessFailedMetrics, tagWorkflow, addAttributes, deleteAttributes, setWorkflowLogToEvent, editWorkflowRouter, getWorkflowSankeyMetrics, getWorkflowRevisionData, getWorkflowRouter, toggleWorkflow, executeWorkflow, getInstancesForWorkflow, getRevisions, getTags, deleteRevision, saveWorkflow, updateWorkflow, discardWorkflow, removeTag} = useWorkflow(Config.url, true, namespace, filepath.substring(1), localStorage.getItem("apikey"))
+    // todo handle err from hook below
+    const {data,  getSuccessFailedMetrics, tagWorkflow, addAttributes, deleteAttributes, setWorkflowLogToEvent, editWorkflowRouter, getWorkflowSankeyMetrics, getWorkflowRevisionData, getWorkflowRouter, toggleWorkflow, executeWorkflow, getInstancesForWorkflow, getRevisions, getTags, deleteRevision, saveWorkflow, updateWorkflow, discardWorkflow, removeTag} = useWorkflow(Config.url, true, namespace, filepath.substring(1), localStorage.getItem("apikey"))
     const [router, setRouter] = useState(null)
 
-    // const [tags, setTags] = useState(null)
+
     const [revisions, setRevisions] = useState(null)
-    const [revsErr, setRevsErr] = useState("")
+    // todo handle revsErr
+    const [, setRevsErr] = useState("")
 
     // fetch revisions using the workflow hook from above
     useEffect(()=>{
@@ -118,7 +120,10 @@ function InitialWorkflowHook(props){
                     :<></>}
                     { activeTab === 1 ?
                         <>
-                        <RevisionSelectorTab tagWorkflow={tagWorkflow} namespace={namespace} filepath={filepath} updateWorkflow={updateWorkflow} setRouter={setRouter} editWorkflowRouter={editWorkflowRouter} getWorkflowRouter={getWorkflowRouter} setRevisions={setRevisions} revisions={revisions} router={router} getWorkflowSankeyMetrics={getWorkflowSankeyMetrics} executeWorkflow={executeWorkflow} getWorkflowRevisionData={getWorkflowRevisionData} searchParams={searchParams} setSearchParams={setSearchParams} deleteRevision={deleteRevision} namespace={namespace} getRevisions={getRevisions} getTags={getTags} removeTag={removeTag} filepath={filepath} />
+                        <RevisionSelectorTab 
+                        tagWorkflow={tagWorkflow}
+                         namespace={namespace}
+                          filepath={filepath} updateWorkflow={updateWorkflow} setRouter={setRouter} editWorkflowRouter={editWorkflowRouter} getWorkflowRouter={getWorkflowRouter} setRevisions={setRevisions} revisions={revisions} router={router} getWorkflowSankeyMetrics={getWorkflowSankeyMetrics} executeWorkflow={executeWorkflow} getWorkflowRevisionData={getWorkflowRevisionData} searchParams={searchParams} setSearchParams={setSearchParams} deleteRevision={deleteRevision}  getRevisions={getRevisions} getTags={getTags} removeTag={removeTag}  />
                         </>
                     :<></>}
                     { activeTab === 2 ?
@@ -485,9 +490,9 @@ function WorkflowInstances(props) {
                             state={obj.node.status} 
                             name={obj.node.as} 
                             id={obj.node.id}
-                            started={dayjs.utc(obj.node.createdAt).local().format("HH:mm a")} 
+                            started={dayjs.utc(obj.node.createdAt).local().format("HH:mm:ss a")} 
                             startedFrom={dayjs.utc(obj.node.createdAt).local().fromNow()}
-                            finished={dayjs.utc(obj.node.updatedAt).local().format("HH:mm a")}
+                            finished={dayjs.utc(obj.node.updatedAt).local().format("HH:mm:ss a")}
                             finishedFrom={dayjs.utc(obj.node.updatedAt).local().fromNow()}
                         />
                     )

--- a/src/layouts/explorer/workflow/revisionTab.js
+++ b/src/layouts/explorer/workflow/revisionTab.js
@@ -494,7 +494,7 @@ export function RevisionSelectorTab(props) {
 
 function TagRevisionBtn(props) {
 
-    let {tagWorkflow, obj, getRevisions, setRevisions, updateTags, getTags, isTag} = props;
+    let {tagWorkflow, obj, getRevisions, setRevisions, updateTags, getTags} = props;
     const [tag, setTag] = useState("")
 
     return(

--- a/src/layouts/instance/index.js
+++ b/src/layouts/instance/index.js
@@ -3,14 +3,14 @@ import './style.css'
 import { Config, copyTextToClipboard } from '../../util';
 import Button from '../../components/button';
 import { useParams } from 'react-router';
-import ContentPanel, { ContentPanelBody, ContentPanelFooter, ContentPanelHeaderButton, ContentPanelHeaderButtonIcon, ContentPanelTitle, ContentPanelTitleIcon } from '../../components/content-panel';
+import ContentPanel, { ContentPanelBody, ContentPanelHeaderButton, ContentPanelHeaderButtonIcon, ContentPanelTitle, ContentPanelTitleIcon } from '../../components/content-panel';
 import FlexBox from '../../components/flexbox';
 import {AiFillCode} from 'react-icons/ai';
 import {useInstance, useInstanceLogs, useWorkflow} from 'direktiv-react-hooks';
 import { CancelledState, FailState, RunningState, SuccessState } from '../instances';
 
 import { Link } from 'react-router-dom';
-import { AutoSizer, List, CellMeasurer, CellMeasurerCache, WindowScroller } from 'react-virtualized';
+import { AutoSizer, List, CellMeasurer, CellMeasurerCache } from 'react-virtualized';
 import { IoCopy, IoEye, IoEyeOff } from 'react-icons/io5';
 import * as dayjs from "dayjs"
 import YAML from 'js-yaml'
@@ -40,13 +40,14 @@ function InstancePage(props) {
     const [wfpath, setWFPath] = useState("")
     const [rev, setRev] = useState("")
     const [follow, setFollow] = useState(true)
-    const [width, setWidth] = useState(window.innerWidth);
+    const [width,] = useState(window.innerWidth);
     const [clipData, setClipData] = useState(null)
     const params = useParams()
 
     let instanceID = params["id"];
 
-    let {data, err, cancelInstance, getInput, getOutput} = useInstance(Config.url, true, namespace, instanceID, localStorage.getItem("apikey"));
+    // todo implement cancelInstance
+    let {data, err,  getInput, getOutput} = useInstance(Config.url, true, namespace, instanceID, localStorage.getItem("apikey"));
 
 
     useEffect(()=>{
@@ -134,12 +135,12 @@ function InstancePage(props) {
                                         ButtonDefinition("Close", () => {}, "small light", true, false)
                                     ]}
                                 >
-                                    <InstanceLogs setClipData={setClipData} clipData={clipData} noPadding namespace={namespace} instanceID={instanceID} follow={follow} setFollow={setFollow} width={width} clipData={clipData} />
+                                    <InstanceLogs setClipData={setClipData} clipData={clipData} noPadding namespace={namespace} instanceID={instanceID} follow={follow} setFollow={setFollow} width={width}/>
                                 </Modal>
                                 </FlexBox>
                             </FlexBox>
                         </ContentPanelTitle>
-                        <InstanceLogs setClipData={setClipData} clipData={clipData} namespace={namespace} instanceID={instanceID} follow={follow} setFollow={setFollow} width={width} clipData={clipData} />
+                        <InstanceLogs setClipData={setClipData} clipData={clipData} namespace={namespace} instanceID={instanceID} follow={follow} setFollow={setFollow} width={width} />
                     </ContentPanel>
                 </FlexBox>
                 <FlexBox className="gap wrap" style={{minHeight: "40%", minWidth: "300px", flex: "2", flexWrap: "wrap-reverse"}}>
@@ -417,28 +418,6 @@ function Output(props){
     )
 }
 
-function InstanceTuple(props) {
-    
-    let {label, value, linkTo} = props;
-
-    let x = value;
-    if (linkTo) {
-        x = (
-            <Link to={linkTo}>{value}</Link>
-        )
-    }
-
-    return (<>
-        <FlexBox className="instance-details-tuple col" style={{minWidth: "150px", flex: "1"}}>
-            <div>
-                <b>{label}</b>
-            </div>
-            <div title={value} style={{fontSize: "12px", overflow: "hidden", textOverflow: "ellipsis", whiteSpace: "nowrap"}}>
-                {x}
-            </div>
-        </FlexBox>
-    </>)
-}
 
 function Logs(props){ 
     const cache = new CellMeasurerCache({

--- a/src/layouts/instances/index.js
+++ b/src/layouts/instances/index.js
@@ -13,7 +13,6 @@ import relativeTime from "dayjs/plugin/relativeTime";
 import utc from "dayjs/plugin/utc"
 import { useNavigate } from 'react-router-dom';
 import Loader from '../../components/loader';
-import { path } from 'd3-path';
 
 dayjs.extend(utc)
 dayjs.extend(relativeTime);
@@ -35,8 +34,8 @@ export default InstancesPage;
 function InstancesTable(props) {
     const {namespace} = props
     const [load, setLoad] = useState(true)
-    const [iQueryParams, setIQueryParams] = useState([])
-    const {data, err, pageInfo} = useInstances(Config.url, true, namespace, localStorage.getItem("apikey"), ...iQueryParams)
+    const [iQueryParams, ] = useState([])
+    const {data, err} = useInstances(Config.url, true, namespace, localStorage.getItem("apikey"), ...iQueryParams)
 
     useEffect(()=>{
         if(data !== null || err !== null) {
@@ -94,9 +93,9 @@ function InstancesTable(props) {
                             state={obj.node.status} 
                             name={obj.node.as} 
                             id={obj.node.id}
-                            started={dayjs.utc(obj.node.createdAt).local().format("HH:mm a")} 
+                            started={dayjs.utc(obj.node.createdAt).local().format("HH:mm:ss a")} 
                             startedFrom={dayjs.utc(obj.node.createdAt).local().fromNow()}
-                            finished={dayjs.utc(obj.node.updatedAt).local().format("HH:mm a")}
+                            finished={dayjs.utc(obj.node.updatedAt).local().format("HH:mm:ss a")}
                             finishedFrom={dayjs.utc(obj.node.updatedAt).local().fromNow()}
                         />
                     )
@@ -121,7 +120,7 @@ const cancelled = "cancelled";
 const running = "pending";
 
 export function InstanceRow(props) {
-    let {state, name, wf, started, startedFrom, finished, finishedFrom, id, namespace} = props;
+    let {state, name, wf, started,  finished,  id, namespace} = props;
     const navigate = useNavigate()
 
     let label;
@@ -155,7 +154,7 @@ export function InstanceRow(props) {
             <div style={{marginLeft:"10px", textOverflow:"ellipsis", overflow:"hidden"}}>
                 /{pathwf.join("/")}
             </div>
-            {pathwf.length != 1 ?
+            {pathwf.length !== 1 ?
             <div>
                 /{wfname}
             </div>:""}

--- a/src/layouts/main/index.js
+++ b/src/layouts/main/index.js
@@ -21,7 +21,6 @@ import GlobalPodPanel from '../global-services/pod'
 import Loader from '../../components/loader';
 import Button from '../../components/button';
 import { IoMenu } from 'react-icons/io5';
-import Login from '../login';
 import InstancePageWrapper from '../instance';
 // import PermissionsPageWrapper from '../permissions';
 import EventsPageWrapper from '../events';
@@ -29,7 +28,7 @@ import Monitoring from '../monitoring';
 
 
 function NamespaceNavigation(props){
-    const {namespaces, namespace, setNamespace, deleteNamespace, deleteErr, extraRoutes, akey} = props
+    const {namespaces, namespace, setNamespace, deleteNamespace, deleteErr, extraRoutes} = props
 
     const [load, setLoad] = useState(true)
     const navigate = useNavigate()

--- a/src/layouts/monitoring/index.js
+++ b/src/layouts/monitoring/index.js
@@ -1,6 +1,6 @@
 import { useNamespaceLogs } from "direktiv-react-hooks"
 import { useEffect, useState } from "react"
-import { VscGraph, VscTerminal } from "react-icons/vsc"
+import {  VscTerminal } from "react-icons/vsc"
 import ContentPanel, { ContentPanelBody, ContentPanelTitle } from "../../components/content-panel"
 import FlexBox from "../../components/flexbox"
 import HelpIcon from "../../components/help"

--- a/src/layouts/namespace-services/index.js
+++ b/src/layouts/namespace-services/index.js
@@ -9,7 +9,7 @@ import FlexBox from "../../components/flexbox";
 import { Config } from "../../util";
 import Modal, { ButtonDefinition, KeyDownDefinition } from "../../components/modal";
 import AddValueButton from "../../components/add-button";
-import {Link, useNavigate} from 'react-router-dom'
+import {Link} from 'react-router-dom'
 import HelpIcon from "../../components/help"
 
 export default function ServicesPanel(props) {

--- a/src/layouts/namespace-services/pod.js
+++ b/src/layouts/namespace-services/pod.js
@@ -148,7 +148,6 @@ export function PodLogs(props){
     const [width, setWidth] = useState(window.innerWidth);
     const [tab, setTab] = useState(pods[0] ? pods[0].name: "")
     const [clipData, setClipData] = useState(null)
-    const params = useParams()
     const [searchParams] = useSearchParams() // removed 'setSearchParams' from square brackets (this should not affect anything: search 'destructuring assignment')
 
     useEffect(()=>{

--- a/src/layouts/settings/broadcast-panel/index.js
+++ b/src/layouts/settings/broadcast-panel/index.js
@@ -1,7 +1,6 @@
 import React from 'react';
 import './style.css';
 import ContentPanel, {ContentPanelTitle, ContentPanelTitleIcon, ContentPanelBody } from '../../../components/content-panel';
-import { BsSliders } from 'react-icons/bs';
 import FlexBox from '../../../components/flexbox';
 import {Config, GenerateRandomKey} from '../../../util';
 import { useBroadcastConfiguration } from 'direktiv-react-hooks';

--- a/src/layouts/settings/variables-panel/index.js
+++ b/src/layouts/settings/variables-panel/index.js
@@ -9,9 +9,7 @@ import { Config } from '../../../util';
 import DirektivEditor from '../../../components/editor';
 import Button from '../../../components/button';
 import {useDropzone} from 'react-dropzone'
-import {BsUpload} from 'react-icons/bs';
 import Tabs from '../../../components/tabs';
-import { RiDeleteBin2Line } from 'react-icons/ri';
 import HelpIcon from '../../../components/help';
 import { VscCloudDownload, VscCloudUpload, VscEye, VscLoading, VscTrash, VscVariableGroup } from 'react-icons/vsc';
 import { AutoSizer } from 'react-virtualized';

